test/e2e/pod_infra_container_test.go

```diff
@@ -435,4 +435,19 @@ var _ = Describe("Podman pod create", func() {
 		Expect(session).Should(Exit(0))
 		Expect(session.OutputToString()).To(ContainSubstring(hostname))
 	})
+
+	tests := []string{"", "none"}
+	for _, test := range tests {
+		It("podman pod create --share="+test+" should not create an infra ctr", func() {
```

Comment:
FYI there must be a `test := test` line above in order to use the correct scope variable.

Reply:
hm really? this worked locally when I ran it, is there a global test variable I'm not seeing?

Reply:
it will likely only test the last element in the slice two times
the for range variable is overwritten each iteration
https://stackoverflow.com/questions/68559574/using-the-variable-on-range-scope-x-in-function-literal-scopelint

Reply:
note that this is racy so it can work but it is not guaranteed to work

Reply:
weird, I'll change that. Thanks for the info I had no idea that was expected behavior

Reply:
see here https://github.com/containers/podman/blob/da1f47921685f40f1b26405278cbf9cb2d06fe09/test/e2e/logs_test.go#L51-L52
https://github.com/containers/podman/pull/12067
we should prevent such nasty and hard to find bugs